src/Foundatio.TestHarness/Caching/HybridCacheClientTestBase.cs

```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Foundatio.AsyncEx;
+using Foundatio.Caching;
+using Foundatio.Messaging;
+using Foundatio.Tests.Extensions;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Foundatio.Tests.Caching;
+
+public class HybridCacheClientTestBase : CacheClientTestsBase, IDisposable
+{
+    protected readonly ICacheClient _distributedCache;
+    protected readonly IMessageBus _messageBus;
+
+    public HybridCacheClientTestBase(ITestOutputHelper output) : base(output)
+    {
+        _distributedCache = new InMemoryCacheClient(o => o.LoggerFactory(Log));
+        _messageBus = new InMemoryMessageBus(o => o.LoggerFactory(Log));
+    }
+
+    /// <summary>
+    /// Returns a hybrid cache client that has a distributed shared cache. For in memory tests, this will be overridden with a shared cache.
+    /// </summary>
+    protected virtual HybridCacheClient GetDistributedHybridCacheClient(bool shouldThrowOnSerializationError = true)
```

Comment:
I don't love this, but this is needed for the in memory hybrid tests that must have a shared distributed client... Also not sure if it makes sense that this class has an instanced of in memory services in the ctor.